rgf/rgf.py

```diff
@@ -516,12 +629,28 @@ def fit(self, X, y):
         np.savetxt(os.path.join(loc_temp, "train.data.x"), X, delimiter=' ', fmt="%s")
         #Store the targets into RGF format
         np.savetxt(os.path.join(loc_temp, "train.data.y"), y, delimiter=' ', fmt="%s")
+        if sample_weight is not None:
+            #Store the weights into RGF format
+            np.savetxt(os.path.join(loc_temp, "train.data.weight"), sample_weight, delimiter=' ', fmt="%s")
+
+        if self.sl2 is None:
+            self.sl2 = self.l2
+
+        if isinstance(self.min_samples_leaf, (float, np.float, np.float16, np.float32, np.float64, np.double)):
```

Comment:
In doc, `self.min_samples_leaf` is float or int.
So do we need np.xx type?

And type of numpy element is `numpy.ndarray`, not `np.float`.
I mean `isinstance(numpy.array(0.5), numpy.ndarray) == True`
If you need to check numpy datatype, you can use `numpy.array(0.5).dtype == numpy.float32`.